### PR TITLE
#28 user management register user command+handler

### LINKED ISSUE #46: RegisterUserCommand + Handler
#28 -  User management feature

- [x]   In Aplication layer create folder Commands
- [x]   In Commands folder  create RegisterNewUser
- [x]   In RegisterNewUser create class RegisterUserCommand
- [x]   In RegisterNewUser create class RegisterUserCommandHandler

### PR DESCRIPTION
**Pull request by BojanMirkovic**
# #28 user management register user command+handler

## Description

  I have creted RegisterUserCommand + Handler, Instaled Serilog for logning and I create IUserRepository just to avoid geting errors during creation of RegisterUserCommandHandler. IUserRepository and UserRepository are not fully implemented

## Changes Made

  I have implement AutoMaper, I faced a lots of small errors during my last PR so I decide to close that one and open new.
  AutoMaper Configuretion moved from program.cs to Application-Layer Dependencyinjection.cs
  Comments are removed from IUserRepository,UserRepository and RegisterUserCommandHandler

## Related Ticket or Issue

- Fixes #46 , #45

## Checklist

- [x] Code adheres to the coding conventions
- [] Unit tests are added for new code
- [] Existing unit tests are updated if needed
- [x] Code follows the principles of Clean Architecture
- [x] Code formatting complies with the `.editorconfig` file
- [x] Pull request template is filled out

### CODE REVIEW

File: API-Layer/Program.cs [hunk 1]
```diff
@@ -16,6 +17,13 @@
     c.SwaggerDoc("v1", new OpenApiInfo { Title = "Dojo BE", Version = "v1" });
 });
 
+// AutoMaper Configuretion
```
Comment (Nemanja1208) [inline API-Layer/Program.cs:20]: Question, if we use Automapper only in Application Layer, why would we reference it here ? We can do that in Dependency Injection in Application layer right ?

File: Application-Layer/Commands/RegisterNewUser/RegisterUserCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        public async Task<UserModel> Handle(RegisterUserCommand request, CancellationToken cancellationToken)
+        {
+            if (request.NewUser == null)
+            {
+                throw new ArgumentNullException("Invalid user data. FirstName,LastName,Email and Password are required.");
+            }
+            try
+            {
+                var userToCreate = _mapper.Map<UserModel>(request.NewUser);
+                userToCreate.Role = "Admin"; // Role is not part of NewUser
```
Comment (Nemanja1208) [inline Application-Layer/Commands/RegisterNewUser/RegisterUserCommandHandler.cs:28]: Remove comment... // Role is not part of NewUser

File: Infrastructure-Layer/Repositories/User/IUserRepository.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using Domain_Layer.Models.UserModel;
+
+namespace Infrastructure_Layer.Repositories.User
+{
+    public interface IUserRepository
+    {
+        Task<UserModel> RegisterUser(UserModel newUser);
+        //Write  rest of IUserRepository methods!
```
Comment (Nemanja1208) [inline Infrastructure-Layer/Repositories/User/IUserRepository.cs:8]: Remove comment

File: Infrastructure-Layer/Repositories/User/UserRepository.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿
+using Domain_Layer.Models.UserModel;
+
+namespace Infrastructure_Layer.Repositories.User
+{
+    public class UserRepository : IUserRepository
+    {
+        // When you are done with writing methods in IUserRepository you can Implement IUserRepository
```
Comment (Nemanja1208) [inline Infrastructure-Layer/Repositories/User/UserRepository.cs:8]: Remove comment

**Review by Nemanja1208 — CHANGES_REQUESTED** [review]
Check comments

File: Infrastructure-Layer/Repositories/User/IUserRepository.cs (hunk 3 above)
Comment (BojanMirkovic) [reply Infrastructure-Layer/Repositories/User/IUserRepository.cs]: done!

File: Application-Layer/Commands/RegisterNewUser/RegisterUserCommandHandler.cs (hunk 2 above)
Comment (BojanMirkovic) [reply Application-Layer/Commands/RegisterNewUser/RegisterUserCommandHandler.cs]: done!

File: API-Layer/Program.cs (hunk 1 above)
Comment (BojanMirkovic) [reply API-Layer/Program.cs]: done!

**Review by BojanMirkovic — COMMENTED** [review]
All comments are resolved ,ready for review

**Review by Nemanja1208 — APPROVED** [review]
GG

File: Application-Layer/Commands/RegisterNewUser/RegisterUserCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        public async Task<UserModel> Handle(RegisterUserCommand request, CancellationToken cancellationToken)
+        {
+            if (request.NewUser == null)
+            {
+                throw new ArgumentNullException("Invalid user data. FirstName,LastName,Email and Password are required.");
+            }
+            try
+            {
+                var userToCreate = _mapper.Map<UserModel>(request.NewUser);
+                userToCreate.Role = "Admin";
```
Comment (kevjo15) [inline Application-Layer/Commands/RegisterNewUser/RegisterUserCommandHandler.cs:28]: Varför tilldelas admin roll till alla nya registerande användare?

**Pull request merged**